tar/bsdtar.c

```diff
@@ -1353,24 +1358,23 @@ configfile_helper(struct bsdtar *bsdtar, const char *line)
 	if ((conf_arg != NULL) && (conf_arg[0] == '~') &&
 	    (bsdtar->homedir != NULL)) {
 		/* Construct expanded argument string. */
-		if (asprintf(&conf_arg_malloced, "%s%s",
+		if (asprintf(&bsdtar->conf_arg, "%s%s",
 		    bsdtar->homedir, &conf_arg[1]) == -1)
 			bsdtar_errc(bsdtar, 1, errno, "Out of memory");
-
-		/* Use the expanded argument string hereafter. */
-		conf_arg = conf_arg_malloced;
 	} else {
-		conf_arg_malloced = NULL;
+		bsdtar->conf_arg = NULL;
 	}
 
 	/* Process the configuration option. */
-	dooption(bsdtar, conf_opt, conf_arg, 1);
+	dooption(bsdtar, bsdtar->conf_opt, bsdtar->conf_arg, 1);
```

Comment:
crap.  No, it can't be strdup, since `conf_arg` can be `NULL`, and `strdup(NULL)` is undefined.  I'll go back to something closer to the original code.